lib/helpers/packageRepoUtils.js

```diff
@@ -11,18 +11,24 @@ class PackageRepoUtils {
     this.registryApiUrl = options.registryApiUrl
       ? options.registryApiUrl
       : NPM_REGISTRY_API
+    this.pkgInfoCache = {}
```

Comment:
this works well now because the Marshall manager (`marshall.js`) instantiates this class once and passes it to the individual marshalls, but if they each instantiated it themselves then it wouldn't maintain the cache because this is reset in the constructor.

what do you think about simply moving this to a const variable in the same file but outside the class, this will make it a true singleton due to how Node.js module system works.

Reply:
Hmm, I'm not sure this is a good idea right now. Moving it outside would mess with the tests, because then we'd have a private variable, for which we don't have proper accessing functions. So for example we couldn't write a test that checks the caching mechanism, since after the first call of  ```getPackageInfo``` in the testfile ```pkginfoCache``` will be set and we'll have no way to clear its value. We could add a ```clearCache``` function or export the cache object, but IMHO none of them are worth it. 
I suggest to leave it as it is, and if we modify the class instantiation logic we'll chose the solution that makes the most sense then.

Reply:
👍